src/pages/index.js

```diff
@@ -143,16 +155,47 @@ const editProfilePopup = new PopupWithForm(
 );
 editProfilePopup.setEventListeners();
 
+const avatarPopup = new PopupWithForm(
+  "#avatar-edit-modal",
+  handleAvatarFormSubmit
+);
+avatarPopup.setEventListeners();
+
 // ---------------- Form Handlers ----------------
 function handleProfileFormSubmit(values) {
   console.log(values);
-  userInfo.setUserInfo({
-    title: values.title,
-    description: values.description,
-  });
-  /* closeModal(profileEditModal); */
-  // closeModal(editProfilePopup);
-  editProfilePopup.close();
+  renderLoading("#profile-edit-modal", true);
+  api
+    .updateUserInfo({
+      name: values.title,
+      about: values.description,
+    })
+    .then((data) => {
+      console.log(data);
+      userInfo.setUserInfo(data);
+      editProfilePopup.close();
+    })
+    .catch((err) => {
+      console.error(`Error updating user info: ${err}`);
+    })
+    .finally(() => {
+      renderLoading("#profile-edit-modal", false);
+    });
+}
+
+function handleAvatarFormSubmit(values) {
+  renderLoading("#avatar-edit-modal", true);
+  api
+    .updateUserAvatar(values.avatar)
+    .then((data) => {
+      userInfo.setUserAvatar(data.avatar);
+      avatarPopup.close();
+      renderLoading("#avatar-edit-modal", false);
```

Comment:
`      renderLoading("#avatar-edit-modal", false);
`

* You need to **only** return the **default** button text in the `finally` block, so that the button could get the text regardless of whether the request succeeds or fails.